narwhal/worker/src/handlers.rs

```diff
@@ -174,146 +170,67 @@ impl<V: TransactionValidator> PrimaryToWorker for PrimaryReceiverHandler<V> {
-                                    WorkerToWorkerClient::new(peer),
-                                    request,
-                                    self.request_batch_timeout,
-                                )
-                            })
-                        }),
-                );
-                debug!(
-                    "Sending BatchRequest retries to workers {names:?}: {:?}",
-                    batch_requests
-                );
             }
-
-            // Fire off batch request(s) and process results. Stop as soon as we have all the
-            // missing batches.
-            while let Some(result) = handles.next().await {
-                match result {
-                    Ok(response) => {
-                        if let Some(batch) = response.into_body().batch {
-                            if !message.is_certified {
-                                // This batch is not part of a certificate, so we need to validate it.
-                                if let Err(err) = self.validator.validate_batch(&batch).await {
-                                    // The batch is invalid, we don't want to process it.
-                                    return Err(anemo::rpc::Status::new_with_message(
-                                        StatusCode::BadRequest,
-                                        format!("Invalid batch: {err}"),
-                                    ));
-                                }
-                            }
-                            let digest = batch.digest();
-                            if missing.remove(&digest) {
-                                self.store.insert(&digest, &batch).map_err(|e| {
-                                    anemo::rpc::Status::internal(format!(
-                                        "failed to write to batch store: {e:?}"
-                                    ))
-                                })?;
-                            }
-                        }
-                        if missing.is_empty() {
-                            return Ok(anemo::Response::new(()));
-                        }
-                    }
-                    Err(e) => {
-                        debug!(
-                            "RequestBatchRequest to worker {:?} failed: {e:?}",
-                            e.peer_id()
-                        )
-                    }
-                }
+            let digest = batch.digest();
+            if missing.remove(&digest) {
+                self.store.insert(&digest, &batch).map_err(|e| {
+                    anemo::rpc::Status::internal(format!("failed to write to batch store: {e:?}"))
+                })?;
             }
+        }
 
-            first_attempt = false;
-            // Add a delay before retrying.
-            sleep(Duration::from_secs(1)).await;
+        if missing.is_empty() {
+            return Ok(anemo::Response::new(()));
         }
+        Err(anemo::rpc::Status::internal("failed to fetch payloads"))
```

Comment:
this is a minor thing, but I wonder if while we are starting to think about cleaning up narwhal, we want to start being more consistent with our naming and use batch consistently instead of a mix of batch and payload?

Reply:
Makes sense.